src/logger.ts

```diff
@@ -0,0 +1,49 @@
+import { createLogger, format, transports } from 'winston';
+import type { TransformableInfo } from 'logform';
+
+const {
+  colorize,
+  combine,
+  errors,
+  printf,
+  simple,
+} = format;
+
+const template = ({
+  level,
+  message,
+  stack,
+  ...rest
+}: TransformableInfo): string => {
+  const meta = Object.keys(rest).length > 0 ? ` ${JSON.stringify(rest)}` : '';
+  const stacktrace = typeof stack === 'string' ? `\n${stack}` : '';
+  const timestamp = new Date().toLocaleString();
+  return `${timestamp} ${level}: ${message}${meta}${stacktrace}`;
+};
+
+const debugLogger = createLogger({
+  format: combine(
+    errors({ stack: true }),
+    colorize(),
+    printf(template),
+  ),
+  level: process.env.LOG_LEVEL ?? 'debug',
+  transports: [
+    new transports.Console(),
```

Comment:
Would be nice if `winston.transports.File` is used instead as opposed to console. In addition a `.logs` folder may be configured and tracked but it's content ignored in `.gitignore`. 

Otherwise, LGTM.

Reply:
Great idea!

Do you think there would be utility on having logging got to console when in a development environment and to a log file in prod, or is that an antipattern?

Reply:
That make sense! Besides you can make that a `.env` setting with a default. However inconvenient/convenient the developer would be able to change it!

Reply:
So it would be added in `.env.example`.

Reply:
Also cc @jasonaowen since he originally set up the logging config that this came from, and may have thoughts to add.

Reply:
My understanding, based on my research at the time, is that saving logs to files is a separate concern that should be handled out-of-application. upload-service and notification-service are started by systemd, and systemd collects their logs and exposes them via journalctl.

If the application manages its own logs, I think we would also need to implement log rotation, configuration for log rotation, configuration for the target files, and probably other things that I'm not thinking of now.

Can you say more about why you want to log to a file instead of the console, @Fenn-CS? What problem are you trying to solve?

Reply:
The logs are useful if someone can see them later, for production what is the "console"? is someone sitting there and watching?

putting the events in files means that they are captured when they happen and someone can see them later. If there's some other mechanism that ensure the same result is achieve then that's great.

However, it very common to see logging being handled as part of the application and not out that sounds like over engineering because they other log levels, like server logs before application logs.

Reply:
So to answer the prod question, I think that console is generally piped to a file by the production deployment mechanism (and that seems to be a pretty standard / expected practice).  Some examples of production logging-to-console being the expected:

* Heroku logs are explicitly [the console output of all processes](https://devcenter.heroku.com/articles/logging)
* Supervisord also defines logging as one of it's primary functions, and in particular it [logs the output of child processes to whatever destination you specify](http://supervisord.org/logging.html).
* Docker seems to [emit logs to stdout and stderr](https://www.datadoghq.com/blog/docker-logging/).

Note: I don't remember what we're using to deploy these things right now, though I know it isn't Docker or Heroku!  But is it Supervisor?  Systemd?  Systemd is probably like supervisor in its log expectations.

---

Curve ball while we think about production logging: what if instead of file logging for production we consider defining a *Sentry* transport stream so logs are written to Sentry?  I think we're trying to use Sentry for Permanent?

---

All of that said: do you think we would want file logging in a dev environment?  I think that's more likely to be used than file-logging-from-application in our production environment, though it does represent complexity / a piece of the system we will have to maintain and update over time, so if console is good enough I think it's reasonable to stick with it.

Additional note: There is a [daily rotate transport](https://github.com/winstonjs/winston-daily-rotate-file) which will do log rotation, but I do agree that there's meaningful simplicity in NOT having our application have to also know how to manage log files.  Then again it's not great to have an application with a built-in time bomb that the engineer needs to actively diffuse (by setting up log rotation manually).

Reply:
I should add this critical point: I think this decision should apply to *all our services* -- we should log in a consistent way; would it be appropriate to move forward on this PR (which mirrors our other services log styles), and to explore how we want service logging to work more broadly?

Reply:
Yes, let's move forward, @slifty .

Reply:
Thanks for the detailed explaination, it all make sense when the a mechanism to manage logging without much effort.

Reply:
Ty all!  And Thank you @Fenn-CS for raising the discussion.